cmd/forwarder/auth_test.go

```diff
@@ -220,7 +220,11 @@ func TestAuthenticate(t *testing.T) {
 				panic(err.Error())
 			}
 			r.SetBasicAuth("user", test.password)
-			assert.Equal(t, test.authenticated, auth.Authenticate(r))
+			if test.authenticated {
+				assert.NotNil(t, auth.Authenticate(r))
```

Comment:
You _could_ put a return after the first assertion, and not include the "else". But 🤷‍♂️

Reply:
@jmervine 
This is a weird case. Normally, I would agree: short-circuiting and less indentation is good. But it just looks weird to me to assert one thing, and then also assert another contrary thing without the if/else block.

I think the code could be improved by including a `credential` field on the test structs, and then doing this:

```go
assert.Equal(t, auth.Credential, auth.Authenticate(r))
```

What do you think of that?

Reply:
@jmervine I fixed it to do ^ :)